### ISSUE TITLE
transformImportsPlugin support mapping

### ISSUE THREAD

**Issue by zhmushan** [web-infra-dev/rspack#5199]
### What problem does this feature solve?

There will always be a few odd imports that don't conform to any rules.

### What does the proposed API of configuration look like?

https://github.com/swc-project/plugins/blob/3d52bc1a6a9e5c78c6f5000d3d8479c471881779/packages/transform-imports/transform/tests/fixture.rs#L59-L60

**Comment by JSerFeng** [comment]
This is likely what you need https://www.rspack.dev/config/builtins.html#builtinspluginimport

**Comment by zhmushan** [comment]
> This is likely what you need https://www.rspack.dev/config/builtins.html#builtinspluginimport

I checked the code, and it seems that rspack's plugin only implements template syntax but does not support mapping. Did I miss something?

**Comment by JSerFeng** [comment]
Not sure what do you mean, what is mapping here ?

**Comment by zhmushan** [comment]
> https://github.com/swc-project/plugins/blob/3d52bc1a6a9e5c78c6f5000d3d8479c471881779/packages/transform-imports/transform/tests/fixture.rs#L59-L60

Referring to the swc test case code I pointed out at the beginning, it **maps** "foo" to "my-library-4/this_is_foo"

**Comment by JSerFeng** [comment]
You can do the same with `library: 'foo', customName: {{ my-library-4/this_is_{ member } }}`

**Comment by zhmushan** [comment]
> You can do the same with `library: 'foo', customName: {{ my-library-4/this_is_{ member } }}`

I understand the template syntax you're talking about, but what I want now **is a mapping similar to SWC**. I need to compute the mapping relationships using JavaScript and provide them for the plugin to use, which can't be fulfilled by the template syntax

**Comment by zhmushan** [comment]
@JSerFeng In the SWC test cases, 'foo' corresponds to 'member' in the rspack plugin, and 'my-library-4' corresponds to 'library'. In the rspack plugin, 'member' can only act as part of a template and nothing else. However, in SWC, it can be mapped to another string

I think I've stated it clearly enough. If you really looked at the test cases I pointed out right from the start (which are just two lines of code), you would understand that the current rspack plugin is not capable of this functionality.

**Let me explain the function of these two lines of code again: they map different members exported from the same library (with consistent modId) to different, unrelated paths.**
